### ISSUE TITLE
no callback to receive reply

### ISSUE THREAD

**Issue by poorcoder** [andrasq/qbean#1]
I am getting this error.  What is my mistake ?

```
node_modules/qbean/lib/qbean.js:263
            throw new Error("no callback to receive reply");
```

My code snippet

```
var channel = 'tweetStream';
bean.use(channel, function(err, using) {
    console.log('channel : ', channel);
    stream.on('tweet', _put);
});

function _put(tweet) {
    var data = {
        user : tweet.user.screen_name,
        text : tweet.text
    };
    var payload = JSON.stringify(data);
    bean.put(0, 0, 10, payload, function(err, jobid) {
        console.log('jobid : ', jobid);
    });     
}
```

**Comment by andrasq** [comment]
Not your mistake, your code looks ok, and when I tried it it worked for me.
Maybe a race condition, or a concurrency issue.

Each command sent to beanstalk expects a matching response, and queues a callback to handle it.
Responses arrive in the same order as the commands were sent, so each response dequeues
the oldest callback and uses it to handle the reply.  The error is saying that there were more
responses received than callbacks queued, which should never happen.  (But it did, so we
have to track down why)

Does the error happen every time or only sometimes?
Does it happen from a single stream or only from multiple streams?
Single node process or clustered?
How large are the payloads?  Perhaps a sockets issue.

**Comment by andrasq** [comment]
could you be connecting multiple beans to the same network connection?
When bean is created, `bean = new QBean({}, stream);`, each bean needs its own
stream, `stream = net.createConnection(11300, beanstalkServer);`

**Comment by poorcoder** [comment]
I am testing on my laptop.
I am getting data from twitter stream api ( https://github.com/ttezel/twit/ )
When first tweet comes, I am getting that error.

```
var Twit = require('twit');
var cnf = require('./config');

// queue
var net = require('net'),
netStream = net.createConnection(11300, 'localhost'),
QBean = require('qbean'),
bean = new QBean({}, netStream);

var channel = 'qbean';
var jobpriority = 0, jobdelay = 0, jobttr = 10;


var params = {
    track: ['test'],
    language: 'tr'
};


var twitter = new Twit(cnf.credentials);

var twitStream = twitter.stream('statuses/filter', params);

twitStream.on('tweet', _put);

function _put(tweet) {
    bean.use(channel, function(err, using) {
    console.log('channel : ', channel);
        var payload = JSON.stringify({
            user : tweet.user.screen_name,
            text : tweet.text
        }); 
        bean.put(jobpriority, jobdelay, jobttr, payload, function(err, jobid) {
            console.log('jobid : ', jobid);
        });
    });
}
```

**Comment by andrasq** [comment]
I think the problem could be the utf8 conversion.  I found two bugs in qbean
relating to sending and receiving multi-byte characters.  The code as originally written
only handled 7-bit ascii because I did not understand some quirks of JSON.stringify()
and Buffer.toString().  Working on a quick patch (base64-encode payload), and will
work on making qbean binary-safe.

**Comment by andrasq** [comment]
fixed in 0.0.7

**Comment by andrasq** [comment]
changed the i/o code to extract byte-counted binary strings directly from the buffers,
wrote unit tests to cover the basics, verified that payloads now support 8-bit binary.
Made a few internal changes to speed things up, and bumped the version to 0.9.2.

**Comment by andrasq** [comment]
changed to read the binary stream with qbuffer, and released as qbean-1.0.0
